### PR TITLE
reduceGraph() helper function for graph routing/trimming

### PR DESCRIPTION
**Pull request by atlas0fd00m**

### CODE REVIEW

File: vivisect/tools/graphutil.py [hunk 1]
```diff
@@ -629,7 +629,16 @@ def clearGraphRouting(graph):
     for nid, ninfo in graph.getNodes():
         graph.getNodeProps(nid)['up'  ] = False
         graph.getNodeProps(nid)['down'] = False
-    
+
+def reduceGraph(graph, props=['up','down']):
+    '''
```
Comment (invisig0th) [inline vivisect/tools/graphutil.py]: you should never, ever, ever, ever initialize default func/method variables with mutable types... :)

File: vivisect/tools/graphutil.py [hunk 2]
```diff
@@ -629,7 +629,16 @@ def clearGraphRouting(graph):
     for nid, ninfo in graph.getNodes():
         graph.getNodeProps(nid)['up'  ] = False
         graph.getNodeProps(nid)['down'] = False
```
Comment (invisig0th) [inline vivisect/tools/graphutil.py]: (as an aside) direct manipulation of the node's property dict is going to be *very* wrong in the new graph API

Comment (atlas0fd00m) [inline vivisect/tools/graphutil.py]: currently (to my knowledge) there is currently no other way to do this in the api

**Comment by atlas0fd00m** [comment]
done.  please recheck.

**Pull request merged**